odoo/addons/base/controllers/avatar.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha1
+
+from odoo.http import Controller, request, route, Response
+
+# show avatar if no image or showAvatar 1
+# show image if image
+
+class Avatar(Controller):
+    @route(['/web/avatar/<int:user_id>', '/web/avatar/<int:user_id>/<string:image_field>'], methods=['GET'])
+    def get_avatar(self, user_id=1, image_field="image_128"):
+        user = request.env['res.users'].browse(user_id).exists()
+        if user:
+            avatar = _generate_avatar_svg(user.name, user.create_date)
+            return Response(avatar, mimetype='image/svg+xml')
+        return Response("User not found.", status=500)
+
+def _generate_avatar_svg(name, create_date):
```

Comment:
Maybe `seed` would be a more suitable name. It doesn't matter for it to be a date, just send the timestamp
```suggestion
def _generate_avatar_svg(name, seed=False):
```